app/controllers/benefits/ReceiveOwnCarMileageBenefitController.scala

```diff
@@ -0,0 +1,121 @@
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.EmploymentSessionService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.{Clock, SessionHelper}
+import views.html.benefits.ReceiveOwnCarMileageBenefitView
+import services.RedirectService._
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class ReceiveOwnCarMileageBenefitController @Inject()(implicit val cc: MessagesControllerComponents,
+                                                      authAction: AuthorisedAction,
+                                                      inYearAction: InYearAction,
+                                                      receiveOwnCarMileageBenefitView: ReceiveOwnCarMileageBenefitView,
+                                                      appConfig: AppConfig,
+                                                      employmentSessionService: EmploymentSessionService,
+                                                      errorHandler: ErrorHandler,
+                                                      ec: ExecutionContext,
+                                                      clock: Clock) extends FrontendController(cc) with I18nSupport with SessionHelper {
+
+
+  def yesNoForm(implicit user: User[_]): Form[Boolean] = YesNoForm.yesNoForm(
+    missingInputError = s"benefits.receiveOwnCarMileageBenefit.error.${if (user.isAgent) "agent" else "individual"}"
+  )
+
+  def show(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
+    inYearAction.notInYear(taxYear) {
+
+      employmentSessionService.getAndHandle(taxYear, employmentId) { (optCya, _) =>
+
+        redirectBasedOnCurrentAnswers(taxYear, employmentId, optCya,
+          EmploymentBenefitsType)(commonCarVanFuelBenefitsRedirects(_, taxYear, employmentId)) { cya =>
+
+          val mileageBenefitQuestion: Option[Boolean] =
+            cya.employment.employmentBenefits.flatMap(_.carVanFuelModel.flatMap(_.mileageQuestion))
+
+          mileageBenefitQuestion match {
+            case Some(questionResult) => Future.successful(Ok(receiveOwnCarMileageBenefitView(yesNoForm.fill(questionResult), taxYear, employmentId)))
+            case None => Future.successful(Ok(receiveOwnCarMileageBenefitView(yesNoForm, taxYear, employmentId)))
+          }
+
+        }
+      }
+    }
+  }
+
+  def submit(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
+    inYearAction.notInYear(taxYear) {
+
+      val redirectUrl: String = CheckYourBenefitsController.show(taxYear, employmentId).url
+
+      employmentSessionService.getSessionDataAndReturnResult(taxYear, employmentId)(redirectUrl) { cya =>
+
+        redirectBasedOnCurrentAnswers(taxYear, employmentId, Some(cya),
+          EmploymentBenefitsType)(commonCarVanFuelBenefitsRedirects(_, taxYear, employmentId)) { cya =>
```

Comment:
as above now that's mileageBenefitsRedirects available can swap to use that, may need to update tests if not all the necessary data is in the test cases like car,van answers etc